modules/custom/activity_creator/src/Controller/NotificationsController.php

```diff
@@ -2,34 +2,98 @@
 
 namespace Drupal\activity_creator\Controller;
 
+use Drupal\activity_creator\ActivityNotifications;
 use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\HtmlCommand;
 use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\views\Views;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
  * Notifications controller.
  */
 class NotificationsController extends ControllerBase {
 
   /**
-   * Ajax callback to mark notifications as read.
+   * The activity notification service.
+   *
+   * @var \Drupal\activity_creator\ActivityNotifications
+   */
+  protected $activities;
+
+  /**
+   * The current user services.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $user;
+
+  /**
+   * NotificationsController constructor.
+   *
+   * @param \Drupal\activity_creator\ActivityNotifications $notifications
+   *   The activity notifications.
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The current user account services.
    */
-  public function readNotificationCallback() {
-    $account = \Drupal::currentUser();
+  public function __construct(ActivityNotifications $notifications, AccountInterface $account) {
+    $this->activities = $notifications;
+    $this->user = $account;
+  }
 
-    // @todo: Add dependency injection.
-    $activity_notifications = \Drupal::service('activity_creator.activity_notifications');
-    $remaining_notifications = $activity_notifications->markAllNotificationsAsSeen($account);
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('activity_creator.activity_notifications'),
+      $container->get('current_user')
+    );
+  }
 
+  /**
+   * Ajax callback to mark notifications as read.
+   *
+   * @deprecated in social:8.x-7.0 and is removed from social:8.x-8.0. Use NotificationsController::getNotificationListCallback()::getNotificationListCallback() instead.
```

Comment:
@jaapjan @ronaldtebrake Do you agree here? : )

Reply:
Yes!